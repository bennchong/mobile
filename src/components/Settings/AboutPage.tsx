import React, { useContext, Component } from "react";
import { ScrollView, Button, View, TouchableOpacity, Alert, Text, Image, StyleSheet } from "react-native";
import { MaterialIcons, AntDesign } from "@expo/vector-icons";
import { withNavigation } from "react-navigation";
import Constants from "expo-constants";
import { styles } from "./styles";
import { StateContext } from "../../state";
import * as Font from 'expo-font';
import Hyperlink from 'react-native-hyperlink';

const imageSource = require("../../assets/rsz_mom-logo-share-hd.png");

interface IAboutPageProps {
  navigation: any;
}

Font.loadAsync({
  'arial-regular': require('../../assets/Fonts/arial.ttf'),
  'calibri-regular': require('../../assets/Fonts/calibri.ttf'),
})

const AboutPage = (props: IAboutPageProps) => {
  const goBack = () => props.navigation.goBack();
  return (
    <ScrollView
      contentContainerStyle={{ marginTop: Constants.statusBarHeight }}
    >
      <View style={{ alignItems: "center", justifyContent: "center" }}>
        <Text style={{fontSize: 16, fontFamily: 'calibri-regular', textAlign: 'center'}}>
          <Text style={{fontWeight: 'bold', fontSize: 24}}>
          {'\n'}{'\n'}SGWorkPass{'\n'}{'\n'}
          </Text>
          <Text style={{color: '#C0C0C0'}}>
            Launched in September 2017{'\n'}
            Version 1.0.8{'\n'}{'\n'}
          </Text>
          Brought to you by{'\n'}
          <Image source={imageSource} style={{ resizeMode: "center" }}/>{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}
          SGWorkPass is a free service by the Singapore{'\n'}Government for users to check the validity of passes{'\n'}issued by the Ministry of Manpower.{'\n'}{'\n'}
          To give us feedback, visit </Text><Hyperlink linkDefault={true} linkStyle={ { color: '#2963b9', fontSize: 16, fontFamily: 'calibri-regular' } }><Text>https://www.mom.gov.sg/contact</Text></Hyperlink><Text>{'\n'}{'\n'}
          {'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}{'\n'}
          <Text style={{color: '#D3D3D3', fontSize: 16, fontFamily: 'calibri-regular', textAlign: 'center'}}>
            &copy; 2019 Ministry of Manpower (MOM).{'\n'}
            All rights reserved.
          </Text>
        </Text>
      </View>
      
      <Button title="Go Back to Settings Page" onPress={goBack} />
    </ScrollView>
  );
};

export default withNavigation(AboutPage);
