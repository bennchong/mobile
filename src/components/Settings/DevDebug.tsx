import React from "react";
import { ScrollView, Button, Text, View } from "react-native";
import { withNavigation } from "react-navigation";
import Constants from "expo-constants";
import { getData } from "@govtechsg/open-attestation";
import { DevDeleteWorkpasses } from "./DevDebugSection/DevDeleteWorkpass";
import { DevStoreWorkPass } from "./DevDebugSection/DevStoreWorkPass";
import { DevStoreDPWorkPassArray } from "./DevDebugSection/DevStoreDPWorkPassArray";
import { useStateValue } from "../../state";
import { DevAppStateText } from "./DevDebugSection/DevAppStateText";
import { VERSION } from "../../config";

interface IDevDebugProps {
  navigation: any;
}

const DevDebug = (props: IDevDebugProps) => {
  const goBack = () => props.navigation.goBack();
  const [{ profilesArray }] = useStateValue();

  let shownWorkpass;
  if (profilesArray[0].workpass === null) {
    shownWorkpass = "Workpass is null";
  } else {
    const cleanWorkpass = getData(profilesArray[0].workpass);
    shownWorkpass = cleanWorkpass.recipient.name;
  }

  return (
    <ScrollView
      contentContainerStyle={{ marginTop: Constants.statusBarHeight }}
    >
      <View style={{ padding: 20 }}>
        <Text>Version: {JSON.stringify(VERSION.revisionId)}</Text>
        <Text> {JSON.stringify(VERSION.publishedTime)} </Text>
      </View>
      <DevStoreWorkPass />
      <DevStoreDPWorkPassArray />
      <DevDeleteWorkpasses />
      <DevAppStateText>Main workpass name: {shownWorkpass}</DevAppStateText>
      <DevAppStateText>
        profilesArray Length: {profilesArray.length}
      </DevAppStateText>
      <DevAppStateText>
        profilesArray: {JSON.stringify(profilesArray)}
      </DevAppStateText>
      <Button title="Go Back to Settings Page" onPress={goBack} />
    </ScrollView>
  );
};

export default withNavigation(DevDebug);
