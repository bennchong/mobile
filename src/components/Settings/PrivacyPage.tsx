import React, { useContext } from "react";
import { ScrollView, Button, View, TouchableOpacity, Alert, Text } from "react-native";
import { MaterialIcons, AntDesign } from "@expo/vector-icons";
import { withNavigation } from "react-navigation";
import Constants from "expo-constants";
import { styles } from "./styles";
import { StateContext } from "../../state";

interface IPrivacyPageProps {
  navigation: any;
}

const PrivacyPage = (props: IPrivacyPageProps) => {
  const goBack = () => props.navigation.goBack();
  return (
    <ScrollView
      contentContainerStyle={{ marginTop: Constants.statusBarHeight }}
    >
      <Text>
          Privacy statement about the Ministry of Manpower's SGWorkPass Mobile App.
      </Text>
      <Button title="Go Back to Settings Page" onPress={goBack} />
    </ScrollView>
  );
};

export default withNavigation(PrivacyPage);
