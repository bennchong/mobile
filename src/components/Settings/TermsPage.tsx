import React from "react";
import { View, StyleSheet, Button } from "react-native";
import { withNavigation } from "react-navigation";
import Constants from "expo-constants";
import PDFReader from 'rn-pdf-reader-js' 

interface IPrivacyPageProps {
  navigation: any;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: Constants.statusBarHeight,
    backgroundColor: '#ecf0f1',
  },
})

const TermsPage = (props: IPrivacyPageProps) => {
  const goBack = () => props.navigation.goBack();
  return (
    <View style={styles.container}>
      <PDFReader
          source={{
            uri: 'https://github.com/sgworkpass/mobile/raw/about_button/src/assets/PDFs/sgworkpass-terms-of-use.pdf',
          }}
        />
      <Button title="Go Back to Settings Page" onPress={goBack} />
    </View>
  );
};

export default withNavigation(TermsPage);
