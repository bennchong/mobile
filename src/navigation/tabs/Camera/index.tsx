import { createStackNavigator } from "react-navigation";
import { ScannerPage } from "../../../components/Camera/ScannerPage";
import ProfilePreviewPage from "../Preview";

const CameraTab = createStackNavigator(
  {
    Home: ScannerPage,
    ProfilePreview: ProfilePreviewPage
  },
  {
    headerMode: "none"
  }
);

export { CameraTab };
