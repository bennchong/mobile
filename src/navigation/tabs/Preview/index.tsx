import { View } from "react-native";
import { withNavigation } from "react-navigation";
import React from "react";
// eslint-disable-next-line no-unused-vars
import { Navigation } from "../../types";
import { ProfileContainer } from "../../../components/profile/ProfileContainer";

interface ProfilePreviewPageProps {
  navigation: Navigation;
}

const ProfilePreviewPage = (props: ProfilePreviewPageProps) => {
  const { navigation } = props;
  const workpass = navigation.getParam("workpass");

  return (
    <View style={{ flex: 1 }}>
      <ProfileContainer
        workpass={workpass}
        navigation={navigation}
        isPreview={true}
      />
    </View>
  );
};

export default withNavigation(ProfilePreviewPage);
