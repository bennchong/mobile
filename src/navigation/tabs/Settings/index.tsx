import { createStackNavigator } from "react-navigation";
import SettingsTemplate from "../../../components/settings/SettingsTemplate";
import DevDebug from "../../../components/settings/DevDebug";

const SettingsTab = createStackNavigator(
  {
    SettingsTemplate,
    DevDebug
  },
  {
    headerMode: "none"
  }
);

export { SettingsTab };
