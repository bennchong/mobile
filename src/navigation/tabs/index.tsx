import { SettingsTab } from "./Settings";
import { CameraTab } from "./Camera";
import { ProfileTab } from "./Profile";

export { SettingsTab, CameraTab, ProfileTab };
