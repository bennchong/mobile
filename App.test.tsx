import React from "react";
import renderer from "react-test-renderer";

import { SettingsTab } from "./src/navigation/tabs/Settings";

describe("<SettingsTab />", () => {
  it("should render without crashing", () => {
    const tree = renderer.create(<SettingsTab />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
