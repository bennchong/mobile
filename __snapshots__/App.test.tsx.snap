// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsTab /> should render without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Settings Page
  </Text>
</View>
`;
